lib/prefab/criteria_evaluator.rb

```diff
@@ -79,14 +77,7 @@ def default_row_values
     end
 
     def in_segment?(criterion, properties)
-      segment_criteria = @resolver.raw(criterion.value_to_match.string)
-
-      # This shouldn't happen, but just in case
-      return false if segment_criteria.nil?
-
-      resolver = CriteriaEvaluator.new(segment_criteria, project_env_id: @project_env_id, resolver: @resolver,
-                                                         base_client: @base_client)
-      resolver.evaluate(properties)
+      @resolver.get(criterion.value_to_match.string, "", properties).bool
```

Comment:
Maybe I'm missing something, but are we actually changing the logic? `@resolver.get` will still call `CriteriaEvaluator#evaluate` on the config it receives, so I think we're effectively doing the same thing as the previous code. See my comment on the tests for additional context/questions.

Or maybe I'm on shrooms and just missing something obvious.

Reply:
The difference here is that the old segment would have criteria.  Those were AND together and tada. 

For a segment that only has a single ConditionalValue and set of criteria, you're right, pretty much the same. 

But if a segment has multiple ConditionalValue we'll now get the benefit of the OR processing that comes from connecting higher up.

Reply:
Ok, that makes sense. 

I think then, my ask for this PR would be to add a test that explicitly demonstrates the new `OR` functionality, as we currently don't have a test for this. I'm happy to add one.

Reply:
@mikowitz please add a test at your leisure 👍

Reply:
Will do! Probably this evening. Watch this space. 💙